src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -612,10 +612,16 @@ private static string[] CreateEnvp(ProcessStartInfo psi)
         {
             var envp = new string[psi.Environment.Count];
             int index = 0;
-            foreach (var pair in psi.Environment)
+            foreach (KeyValuePair<string, string?> pair in psi.Environment)
             {
-                envp[index++] = pair.Key + "=" + pair.Value;
+                // Ignore null values for consistency with Environment.SetEnvironmentVariable
+                if (pair.Value != null)
```

Comment:
I am waiting to see first whether there are existing tests that hit this case, and then add as necessary.